src/web/routes/application/middleware/handle-path-request.js

```diff
@@ -26,10 +28,15 @@ const middleware = (config, pathsInSequence) => (req, res, next) => {
     return res.redirect(nextAllowedPath)
   }
 
+  // If step is not navigable then return to the next allowed path (which will be the originating page)
```

Comment:
I don't think this will always be the originating page - the referring path and `nextAllowedPath` could be different